CruCentralCoast/Community/CommunityGroups/CommunityGroupsVC.swift

```diff
@@ -48,29 +72,163 @@ class CommunityGroupsVC: UITableViewController {
+        }
+        
+        return 0
+
     }
     
     override func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
         let cell = tableView.dequeueCell(CommunityGroupCell.self, indexPath: indexPath)
-        
-        cell.configure(with: self.dataArray[indexPath.row])
-
+        if isFiltering() {
+            // check if there's item(s) for that day
+            if let filteredDay = self.filteredDictionary[self.filteredDays[indexPath.section]] {
+                cell.configure(with: filteredDay[indexPath.row])
+            }
+        }
+        else {
+            // check if there's item(s) for that day
+            if let day = self.dataDictionary[self.days[indexPath.section]] {
+                cell.configure(with: day[indexPath.row])
+            }
+        }
+         
         return cell
     }
     
     override func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
         let vc = UIStoryboard(name: "CommunityGroups", bundle: nil).instantiateViewController(CommunityGroupDetailsVC.self)
-        vc.configure(with: self.dataArray[indexPath.row])
+        if isFiltering() {
+            if let filteredDay = self.filteredDictionary[self.filteredDays[indexPath.section]] {
+                vc.configure(with: filteredDay[indexPath.row])
+            }
+        }
+        else {
+            if let day = self.dataDictionary[self.days[indexPath.section]] {
+                vc.configure(with: day[indexPath.row])
+            }
+        }
+        
         self.navigationController?.pushViewController(vc, animated: true)
     }
+    
+    
+    
+    // MARK: - Private instance methods
+    
+    func searchBarIsEmpty() -> Bool {
+        // Returns true if the text is empty or nil
+        return searchController.searchBar.text?.isEmpty ?? true
+    }
+    
+    func isFiltering() -> Bool {
+        return searchController.isActive && !searchBarIsEmpty()
+    }
+    
+    func filterContentForSearchText(_ searchText: String, scope: String = "All") {
+        
+        self.filteredDays.removeAll()
+        self.filteredArray = self.dataArray.filter({ (communityGroup : CommunityGroup) -> Bool in
+            
+            // specified search criteria to filter
+            let searchArray = [communityGroup.gender.rawValue,
+                               communityGroup.leaderNames,
+                               communityGroup.time,
+                               communityGroup.weekDay.rawValue,
+                               communityGroup.year.rawValue]
+            for category in searchArray {
+                if category?.lowercased().contains(searchText.lowercased()) ?? false {
+                    return true
+                }
+            }
+            return false
+        })
+        
+        self.filteredDictionary = self.filteredArray.toDictionary { $0.weekDay }
+        for day in self.days {
+            if self.filteredDictionary[day] != nil {
```

Comment:
You could probably reduce this to something like `self.days.filter({ self.filteredDictionary[$0] != nil })`